include/zephyr/pm/policy.h

```diff
@@ -137,38 +137,37 @@ void pm_policy_state_lock_put(enum pm_state state, uint8_t substate_id);
  */
 bool pm_policy_state_lock_is_active(enum pm_state state, uint8_t substate_id);
 
-
 /**
  * @brief Register an event.
  *
  * Events in the power-management policy context are defined as any source that
  * will wake up the system at a known time in the future. By registering such
  * event, the policy manager will be able to decide whether certain power states
  * are worth entering or not.
- * CPU is woken up before the time passed in cycle to prevent the event handling
- * latency
  *
- * @note It is mandatory to unregister events once they have happened by using
- * pm_policy_event_unregister(). Not doing so is an API contract violation,
- * because the system would continue to consider them as valid events in the
- * *far* future, that is, after the cycle counter rollover.
+ * CPU is woken up before the time passed in cycle to minimize event handling
+ * latency. Once woken up, the CPU will be kept awake until the event has been
+ * handled, which is signaled by pm_policy_event_unregister().
```

Comment:
I don't see anything in this code keeping the CPU awake. What am I missing?

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/84532b24e54e9ba205a429f004070e59e6c0105d/subsys/pm/pm.c#L151-L156

Reply:
CPU is kept alive if:
* There is a PM policy event within CPU residency time
* There is a kernel event (timeout expiration) within CPU residency time.

Reply:
The linked code snippet wakes up the CPU before the event. But is there anything keeping it up? Is something blocking the idle thread from entering sleep again until `pm_policy_event_unregister()` is called?

Reply:
yes, if an event is active, meaning it needs to be unregistered/handled, pm_policy_next_event_ticks() returns 0, which means there is an event in 0 ticks (AKA right now), which will prevent the CPU from sleeping as the time until next event is 0.

Reply:
I see. To reduce ambiguity, I moved the discussion to https://github.com/zephyrproject-rtos/zephyr/pull/80580#discussion_r1869519589